tl/features/context_match.py

```diff
@@ -1,32 +1,51 @@
 import pandas as pd
 import re
 import rltk.similarity as similarity
+from tl.exceptions import RequiredInputParameterMissingException
+from statistics import mode
+import gzip
 
 
 class MatchContext(object):
-    def __init__(self, input_path, context_path, args):
+    def __init__(self, input_path, args, context_path=None, custom_context_path=None):
         self.final_data = pd.read_csv(input_path, dtype=object)
         self.data = pd.DataFrame()
         self.result_data = pd.DataFrame()
-        self.context = self.read_context_file(context_path)
+        self.isCustom = False
+        if context_path is None and custom_context_path is None:
+            raise RequiredInputParameterMissingException(
+                'One of the input parameters is required: {} or {}'.format("context_path", "custom_context_path"))
+        elif custom_context_path:
+            self.isCustom = True
+            self.context = self.read_context_file(custom_context_path)
+        else:
+            self.context = self.read_context_file(context_path)
         self.output_column_name = args.get("output_column")
         self.similarity_string_threshold = args.pop("similarity_string_threshold")
         self.similarity_quantity_threshold = args.pop("similarity_quantity_threshold")
+        self.string_separator = args.pop("string_separator")
+        self.string_separator = self.string_separator.replace('"', '')
+        self.properties_with_score_metric = pd.DataFrame(columns=['property', 'position', 'value', 'min_sim'])
         self.debug = args.pop("debug")
 
-    @staticmethod
-    def read_context_file(context_path: str) -> dict:
+    def read_context_file(self, context_path: str) -> dict:
         context_dict = {}
-
-        f = open(context_path)
+        if self.isCustom:
+            f = gzip.open("coauthors.context.tsv.gz", 'rt')
+            column_1 = "node1"
+            column_2 = "node2"
+        else:
+            f = open(context_path)
+            column_1 = "qnode"
```

Comment:
again talking about parameter names, why not call these node1_column and node2_column ? Please change this